File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/EmojiChangedEvent.java
package work.socialhub.kslack.entity.event;

import java.util.List;

/**
 * The email_domain_changed event is sent to all connections for a workspace
 * when the email domain settings for a workspace change.
 * Most clients can ignore this event.
 * <p>
 * https://api.slack.com/events/emoji_changed
 */
public class EmojiChangedEvent implements Event {

    public static final String TYPE_NAME = "emoji_changed";

    private final String type = TYPE_NAME;

    private String subtype; // possible values: add, remove

    private List<String> names; // only for subtype:remove

    private String name; // only for subtype:add

    /**
     * The URL of the image
     */
    private String value; // only for subtype:add

    private String eventTs;

    public String getType() {
        return this.type;
    }

    public String getSubtype() {
        return this.subtype;
    }

    public List<String> getNames() {
        return this.names;
    }

    public String getName() {
        return this.name;
    }

    public String getValue() {
        return this.value;
    }

    public String getEventTs() {
        return this.eventTs;
    }

    public void setSubtype(String subtype) {
        this.subtype = subtype;
    }

    public void setNames(List<String> names) {
        this.names = names;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setValue(String value) {
        this.value = value;
    }

    public void setEventTs(String eventTs) {
        this.eventTs = eventTs;
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/EmojiChangedEvent.kt
package work.socialhub.kslack.entity.event

/**
 * The email_domain_changed event is sent to all connections for a workspace
 * when the email domain settings for a workspace change.
 * Most clients can ignore this event.
 *
 *
 * https://api.slack.com/events/emoji_changed
 */
class EmojiChangedEvent : Event {
    override val type: String = TYPE_NAME

    override var subtype: String? = null // possible values: add, remove

    var names: Array<String>? = null // only for subtype:remove

    var name: String? = null // only for subtype:add

    /**
     * The URL of the image
     */
    var value: String? = null // only for subtype:add

    var eventTs: String? = null

    companion object {
        const val TYPE_NAME: String = "emoji_changed"
    }
}